### PR TITLE
Add default timeout for lb client

### LINKED ISSUE #5642: Lack of timeout in client calls causing 15m53s hang in update of load balancers
#### What happened:

We have been observing a lock up in the CCM that lasts for exactly 15m53s repeatedly through the OpenShift testing infrastructure. This has been happening approximately since we started using the 1.29 release of the CCM.

Digging into the detail, it happens specifically when updating the load balancer. Through instrumenting (with additional logging) I have managed to reproduce the issue and initially narrowed the hang down to [CreateOrUpdateLoadBalancer](https://github.com/openshift/cloud-provider-azure/blob/b8d9174689fb3ba103f5c4ea1464337c93e3a759/pkg/azureclients/loadbalancerclient/azure_loadbalancerclient.go#L224-L252) in the `azureclients` pacakge.

Instrumenting more narrowed it specifically to [createOrUpdateLoadBalancer](https://github.com/openshift/cloud-provider-azure/blob/b8d9174689fb3ba103f5c4ea1464337c93e3a759/pkg/azureclients/loadbalancerclient/azure_loadbalancerclient.go#L254-L283). The function itself does very little except a `PUT` call out to the Azure APIs.

It seems that the call is heading out and the future is never getting cancelled, meaning we hang until something else happens, some sort of background reset?

This code is called from [CreateOrUpdateLB](https://github.com/openshift/cloud-provider-azure/blob/7c0a8afb84df58905abe03583929f0f70ad48236/pkg/provider/azure_loadbalancer_repo.go#L135-L141) in the provider package, and if you look at that, you can see that it's passing a `context.Background()` through. 

There should be a timeout on all contexts passed to asynchnronous calls like this, but it appears it is missing here.

#### What you expected to happen:

All calls to external APIs should timeout in a reasonable timeframe and should not hang indefinitely.

#### How to reproduce it (as minimally and precisely as possible):

Seems sporadic, but we are seeing it approximately half the time in our E2E testing.

#### Environment:

- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by MartinForReal**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Add default timeout for lb client
#### Which issue(s) this PR fixes:

Fixes #5642

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/provider/azure_loadbalancer_repo.go [hunk 1]
```diff
@@ -40,8 +40,7 @@ import (
 
 // DeleteLB invokes az.LoadBalancerClient.Delete with exponential backoff retry
 func (az *Cloud) DeleteLB(service *v1.Service, lbName string) *retry.Error {
-	ctx, cancel := getContextWithCancel()
-	defer cancel()
+	ctx, _ := context.WithTimeout(context.Background(), 60*time.Second)
```
Comment (feiskyer) [inline pkg/provider/azure_loadbalancer_repo.go:43]: why getting the cancel and following `defer cancel()` removed?

Comment (MartinForReal) [reply pkg/provider/azure_loadbalancer_repo.go:43]: getContextWithCancel returned a ctx whose parent context is background,  context.WithTimeout did the same thing

**Comment by JoelSpeed** [comment]
Yeah I think this will help, thanks
/lgtm

**Comment by feiskyer** [comment]
@MartinForReal could you cherry-pick to 1.27 branch manually?

**Pull request merged**